client/src/components/Follow.tsx

```diff
@@ -1,43 +1,60 @@
-import React, { ReactElement, useState, useEffect } from 'react';
+import React, { ReactElement, useState, useEffect, useContext } from 'react';
 import { useParams } from 'react-router-dom';
+import { UserContext } from './UserContext';
 import axios from 'axios';
+import Button from '@mui/material/Button';
 
 const Follow = (): ReactElement => {
+  const userId = useContext(UserContext);
   const { id } = useParams();
   const [isFollowing, setIsFollowing] = useState(false);
 
   useEffect(() => {
     axios.get(`/api/follows/isFollowing/${id}`).then(({ data }) => {
       setIsFollowing(data.isFollowing);
-    }).catch;
+    });
   }, []);
 
   const follow = () => {
-    axios
-      .post(`/api/follows/follow/${id}`)
-      .then((): void => {
-        setIsFollowing(true);
-      })
-      .catch((err) => console.error(err));
+    axios.post(`/api/follows/follow/${id}`).then((): void => {
+      if (userId === Number(id)) {
+        return;
+      }
+      setIsFollowing(true);
+    });
```

Comment:
Could you explain what this is doing? I don't understand it. Also you could just check for `if (userId !== +id) { setIsFollowing(true);}